test/powershell/Language/Scripting/OutErrorVariable.Tests.ps1

```diff
@@ -0,0 +1,390 @@
+﻿function Validate-Result {
+    param($out, $outStr, $count, $v)
+
+    It "$outStr.count" { $out.count | Should Be $count }
+    for($i=0; $i -lt $count; $i++)
+    {
+        It "$outStr[$i]" { $out[$i] | Should Be $v[$i] } 
+    }                
+}
+
+Describe "Tests OutVariable only" -Tags "CI" {
+    BeforeAll {
+
+        function get-foo1
+        {
+            [CmdletBinding()]
+            param()
+
+            "foo"
+        }
+
+        function get-foo2
+        {
+            [CmdletBinding()]
+            param()
+
+            $pscmdlet.writeobject("foo")
+        }
+
+        function get-bar 
+        {
+            [CmdletBinding()]
+            param()
+
+            "bar"
+            get-foo1 -outVariable global:a
+        }
+    }
+
+    Context 'Updating OutVariable Case 1: pipe string' {        
+
```

Comment:
These aren't going to do anything because they're not in it blocks, also these really look like they could all be done via test cases

```

$testcases = @{ Title = "Updating OutVariable Case 1: pipe string"; command = "get-foo1"; OutVariable = "a"; Expected = "foo" },
        @{ Title = 'Updating OutVariable Case 2: $pscmdlet.writeobject'; command = "get-foo1"; OutVariable = "a"; Expected = "foo" }

Describe Tests {
    BeforeAll {
    
        function get-foo1
        {
            [CmdletBinding()]
            param()

            "foo"
        }

        function get-foo2
        {
            [CmdletBinding()]
            param()

            $pscmdlet.writeobject("foo")
        }

        function get-bar 
        {
            [CmdletBinding()]
            param()

            "bar"
            get-foo1 -outVariable global:a
        }

    }
    It "<Title>" -TestCases $testcases {
        param ( $command, $arguments, $Expected, $OutVariable )
        & $command -OutVariable $OutVariable > $null
        Get-Variable -ValueOnly $outVariable | should be $Expected
    }

}
```